File: dto/src/main/java/com/krillsson/sysapi/dto/gpu/GpuLoad.java
package com.krillsson.sysapi.dto.gpu;

public class GpuLoad {
    private double coreLoad;
    private double memoryLoad;
    private GpuHealth health;

    public GpuLoad(double coreLoad, double memoryLoad, GpuHealth health) {
        this.coreLoad = coreLoad;
        this.memoryLoad = memoryLoad;
        this.health = health;
    }

    public GpuLoad() {
    }

    public double getCoreLoad() {
        return coreLoad;
    }

    public void setCoreLoad(double coreLoad) {
        this.coreLoad = coreLoad;
    }

    public double getMemoryLoad() {
        return memoryLoad;
    }

    public void setMemoryLoad(double memoryLoad) {
        this.memoryLoad = memoryLoad;
    }

    public GpuHealth getHealth() {
        return health;
    }

    public void setHealth(GpuHealth health) {
        this.health = health;
    }
}

File: sys-api-spring/src/main/kotlin/com/krillsson/sysapi/core/history/db/GpuLoad.kt
package com.krillsson.sysapi.core.history.db

import jakarta.persistence.Embeddable
import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.util.UUID

@Entity
class GpuLoad(
    @Id
    open var id: UUID,
    @JoinColumn(name = "historyId", insertable = false, updatable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    open var history: HistorySystemLoadEntity? = null,
    open var historyId: UUID,
    open var name: String,
    open var coreLoad: Double,
    open var memoryLoad: Double,
    @Embedded
    open var health: GpuHealth
)

@Embeddable
class GpuHealth(open var fanRpm: Double, open var fanPercent: Double, open var temperature: Double)